community/bolt/src/main/java/org/neo4j/bolt/v1/runtime/internal/CypherAdapterStream.java

```diff
@@ -81,6 +85,12 @@ public boolean visit( Result.ResultRow row ) throws Exception
             String metadataFieldName = rootPlanTreeNode.hasProfilerStatistics() ? "profile" : "plan";
             visitor.addMetadata( metadataFieldName, ExecutionPlanConverter.convert( rootPlanTreeNode ) );
         }
+
+        Iterable<Notification> notifications = delegate.getNotifications();
+        if ( notifications.iterator().hasNext() )
+        {
+            visitor.addMetadata( "notifications", NotificationConverter.convert( notifications ) );
+        }
```

Comment:
I'd pull the `notifications.iterator()` into a local variable, because I think the `iterator()` is supposed to return a new Iterator every time.